### PR TITLE
[executor] create ExecutedTrees to encapsulate both state tree and tx…

### PR DESCRIPTION
**Pull request by lightmark**
…n accumulator

## Motivation

Create a struct `ExecutedTrees` to encapsulate the two trees amended in speculation.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Y

## Test Plan

CI

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/libra/website, and link to your PR here.)

### CODE REVIEW

File: execution/executor/src/lib.rs [hunk 1]
```diff
@@ -297,3 +303,35 @@ enum Command {
         resp_sender: oneshot::Sender<Result<ExecuteChunkResponse>>,
     },
 }
+
+#[derive(Clone, Debug)]
+pub struct ExecutedTrees {
+    /// The in-memory Sparse Merkle Tree representing a specific. If this tree is presenting the
```
Comment (zamsden) [inline execution/executor/src/lib.rs]: "specific execution"

File: execution/executor/src/block_processor.rs [hunk 2]
```diff
@@ -694,14 +670,13 @@ where
...
-        previous_state_tree: Rc<SparseMerkleTree>,
-        previous_transaction_accumulator: Rc<Accumulator<TransactionAccumulatorHasher>>,
+        parent_trees: &ExecutedTrees,
     ) -> Result<ProcessedVMOutput> {
         // The data of each individual transaction. For convenience purpose, even for the
         // transactions that will be discarded, we will compute its in-memory Sparse Merkle Tree
         // (it will be identical to the previous one).
         let mut txn_data = vec![];
-        let mut current_state_tree = previous_state_tree;
+        let mut current_state_tree = parent_trees.state_tree().clone();
```
Comment (msmouse) [inline execution/executor/src/block_processor.rs]: nit: it's more ergonomic to do `Rc::clone(parent_trees.state_tree())`

File: storage/storage_client/src/state_view.rs [hunk 3]
```diff
@@ -81,18 +81,18 @@ impl<'a> VerifiedStateView<'a> {
...
         Self {
             reader,
-            latest_persistent_version: if num_elements_in_accumulator > 0 {
-                Some(num_elements_in_accumulator - 1)
-            } else {
-                None
-            },
-            latest_persistent_state_root,
+            latest_persistent_version: latest_persistent_version_and_state_root.0,
+            // if num_elements_in_accumulator > 0 {
```
Comment (wqfish) [inline storage/storage_client/src/state_view.rs]: ???

Comment (lightmark) [reply storage/storage_client/src/state_view.rs]: forget this.

**Pull request closed without merging**